forester/src/send_transaction.rs

```diff
@@ -0,0 +1,388 @@
+                )
+                .await;
+            num_sent_transactions += transactions.len() as u64;
+
+            info!("build transaction time {:?}", start_time.elapsed());
+            let start_time_get_connections = tokio::time::Instant::now();
+            info!(
+                "get get connections txs time {:?}",
+                start_time_get_connections.elapsed()
+            );
+            let url = (*rpc).get_url().to_string();
+            // 6. Spawn new thread to build and send transactions.
+            // Will time out with retry_config.max_retries or retry_config.global_timeout.
+            tokio::spawn(async move {
+                let rpc = SolanaRpcConnection::new(url, None);
+                let mut results = vec![];
+                for transaction in transactions.iter() {
+                    let res = send_signed_transaction(transaction, &rpc, config.retry_config);
+                    results.push(res);
+                }
+                let all_results = join_all(results);
+                all_results.await;
+            });
+
+            info!(
+                "get send txs time {:?}",
+                start_time_get_connections.elapsed()
+            );
+            // 7. Await minimum batch time.
+            batch_min_time.await;
+        }
+
+        // 8. If work items is empty, await minimum batch time.
+        // If this is triggered we could switch to subscribing to the queue
+        if work_items.is_empty() {
+            info!("Work items empty, waiting for next batch epoch {:?}", epoch);
+            tokio::time::sleep(batch_time).await;
+        }
+    }
+    Ok(num_sent_transactions)
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct SendBatchedTransactionsConfig {
+    pub num_batches: u64,
+    pub batch_time_ms: u64,
+    pub build_transaction_batch_config: BuildTransactionBatchConfig,
+    pub retry_config: RetryConfig,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct BuildTransactionBatchConfig {
+    pub batch_size: u64,
+    pub compute_unit_price: Option<u64>,
+    pub compute_unit_limit: Option<u32>,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct RetryConfig {
+    pub max_retries: u8,
+    pub retry_wait_time_ms: u64,
+    pub global_timeout: u128,
+}
+
+/// Sends a transaction and retries if not confirmed after retry wait time.
+/// Stops retrying at the global timeout (end of light slot).
+pub async fn send_signed_transaction(
+    transaction: &Transaction,
+    connection: &SolanaRpcConnection,
+    config: RetryConfig,
+) -> Result<Signature> {
+    let mut retries = 0;
+    let retry_wait_time = Duration::from_millis(config.retry_wait_time_ms);
+    let txid = transaction.signatures[0];
+    while retries < config.max_retries && get_current_system_time_ms() < config.global_timeout {
+        match connection.send_transaction(transaction).await {
+            Ok(_signature) => {
+                // info!("Transaction sent: {}", signature);
```

Comment:
Still needed?

I wouldn't mind keeping it around, maybe we could just change `info!` to `debug!`